framework/src/multiapps/TransientMultiApp.C

```diff
@@ -395,8 +445,61 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
             }
           }
         }
-        else if (!ex->lastSolveConverged())
-          throw MultiAppSolveFailure(name() + " failed to converge");
+        else
+        {
+          if (!ex->lastSolveConverged())
+          {
+            // Even if we don't allow auto_advance - we can still catch up to the current time if
+            // possible
+            if (_catch_up)
+            {
+              _console << "Starting Catch Up!" << std::endl;
+
+              bool caught_up = false;
+
+              unsigned int catch_up_step = 0;
+
+              Real catch_up_dt = dt / 2;
+
+              while (!caught_up && catch_up_step < _max_catch_up_steps)
```

Comment:
`caught_up` can never be true here. I suspect that you are putting this here "just in case" but maybe it should just be an assertion instead to protect from refactoring issues later?

Reply:
Fair enough - it was there from the copy/paste from above where it would turn true to end the loop (but in this version it breaks instead).  I'll remove it.